arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift

```diff
@@ -0,0 +1,117 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class NearestVertexViewController: UIViewController {
+    /// Create the graphics and symbology for the tapped point, the nearest vertex and the nearest coordinate.
+    let tappedLocationSymbol = AGSSimpleMarkerSymbol(style: .X, color: .orange, size: 15)
+    let nearestCoordinateSymbol = AGSSimpleMarkerSymbol(style: .diamond, color: .red, size: 10)
+    let nearestVertexSymbol = AGSSimpleMarkerSymbol(style: .circle, color: .blue, size: 15)
+    
+    /// The symbology for the example polygon area.
+    let polygonFillSymbol = AGSSimpleFillSymbol(style: .forwardDiagonal, color: .green, outline: AGSSimpleLineSymbol(style: .solid, color: .green, width: 2))
+    
+    /// The graphics overlay for the polygon and points..
+    let graphicsOverlay = AGSGraphicsOverlay()
+    
+    /// Create the point collection that defines the polygon.
+    let createdPolygon: AGSPolygon = {
+        let polygonBuilder = AGSPolygonBuilder(spatialReference: .webMercator())
+        polygonBuilder.addPointWith(x: -5991501.677830, y: 5599295.131468)
+        polygonBuilder.addPointWith(x: -6928550.398185, y: 2087936.739807)
+        polygonBuilder.addPointWith(x: -3149463.800709, y: 1840803.011362)
+        polygonBuilder.addPointWith(x: -1563689.043184, y: 3714900.452072)
+        polygonBuilder.addPointWith(x: -3180355.516764, y: 5619889.608838)
+        return polygonBuilder.toGeometry()
+    }()
+    
+    /// The graphic for the polygon, tapped point, nearest coordinate point and nearest vertex point.
+    lazy var polygonGraphic = AGSGraphic(geometry: createdPolygon, symbol: polygonFillSymbol)
+    lazy var tappedLocationGraphic = AGSGraphic(geometry: nil, symbol: tappedLocationSymbol)
+    lazy var nearestCoordinateGraphic = AGSGraphic(geometry: nil, symbol: nearestCoordinateSymbol)
+    lazy var nearestVertexGraphic = AGSGraphic(geometry: nil, symbol: nearestVertexSymbol)
+    
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+            mapView.graphicsOverlays.add(graphicsOverlay)
+            mapView.setViewpointCenter(AGSPoint(x: -4487263.495911, y: 3699176.480377, spatialReference: .webMercator()), scale: 1e8)
+            mapView.touchDelegate = self
+        }
+    }
+    
+    /// Creates a map.
+    ///
+    /// - Returns: A new `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .topographic())
+        return map
+    }
+    
+    /// Add the graphics to the graphics overlay.
+    func addGraphicsToOverlay() {
+        graphicsOverlay.graphics.add(polygonGraphic)
+        graphicsOverlay.graphics.add(nearestCoordinateGraphic)
+        graphicsOverlay.graphics.add(tappedLocationGraphic)
+        graphicsOverlay.graphics.add(nearestVertexGraphic)
```

Comment:
It would probably be more efficient to add the graphics all at once rather than one by one:

```suggestion
        graphicsOverlay.graphics.addObjects(from: [
            polygonGraphic,
            nearestCoordinateGraphic,
            tappedLocationGraphic,
            nearestVertexGraphic,
        ])
```